ST_seedDispersal.c

```diff
@@ -0,0 +1,324 @@
+{
+	int senderRow, senderCol, recieverRow, recieverCol;
+	SppIndex sp;
+	double MAXD; /* maximum seed dispersal distance for a given species */
+	double maxRate; /* Dispersability of the seeds */
+	double plotWidth; /* width of the plots (all plots are equal and square) */
+	double distanceBetweenPlots; /* distance between the sender and the reciever */
+    CellType* sender;
+
+    RandSeed(SuperGlobals.randseed, &dispersal_rng);
+
+	/* sender denotes that these loops refer to the cell distributing seeds */
+	for(senderRow = 0; senderCol < grid_Rows; ++senderRow){
+		for(senderCol = 0; senderCol < grid_Cols; ++senderCol){
+			/* Cell is loaded to ensure the ForEachSpecies loop works */
+			load_cell(senderRow, senderCol);
+            sender = &gridCells[senderRow][senderCol];
+
+			/* Allocate seed dispersal information. */
+			sender->mySeedDispersal = Mem_Calloc(MAX_SPECIES, sizeof(Grid_SD_St), "initDispersalParameters");
+
+			ForEachSpecies(sp){
+				if (!(Species[sp]->use_me && Species[sp]->use_dispersal)){
+					continue;
+				}
+
+				/* These are the three values we need to calculate the probability of dispersal */
+				MAXD = ((Species[sp]->sd_H * Species[sp]->sd_VW) / Species[sp]->sd_VT) / 100.0; // divided by 100 to convert from cm to m.
+				maxRate = -(log(0.005) / MAXD);
+				plotWidth = sqrt(Globals->plotsize);
+
+				/* Allocate the dispersalProb 2d array */
+				sender->mySeedDispersal[sp].dispersalProb = Mem_Calloc(grid_Rows, 
+							sizeof(double*), "initDispersalParameters: dispersalProb");
+				for(recieverRow = 0; recieverRow < grid_Rows; ++recieverRow){
+					sender->mySeedDispersal[sp].dispersalProb[recieverRow] = 
+								Mem_Calloc(grid_Cols, sizeof(double), "initDispersalParameters: dispersalProb[i]");
+				}
+
+				/* Loop through all possible recipients of seeds. */
+				for(recieverRow = 0; recieverRow < grid_Rows; ++recieverRow){
+					for(recieverCol = 0; recieverCol < grid_Cols; ++recieverCol){
+						if(senderRow == recieverRow && senderCol == recieverCol){
+							continue; // No need to calculate a probability for dispersal to itself
+						}
+                        
+						distanceBetweenPlots = _cell_dist(senderRow, recieverRow, senderCol, recieverCol, plotWidth);
+
+                        /* The value that we are after should be saved to the sender cell. */
+						sender->mySeedDispersal[sp].dispersalProb[recieverRow][recieverCol]
+                                    = (distanceBetweenPlots > MAXD) ? (0.0) : (exp(-maxRate * distanceBetweenPlots));
+					}
+				}
+			}
+		}
+	}
+	unload_cell();
+}
+
+/* Perform seed dispersal durring the simulation. This is NOT functional yet. */
+void disperseSeeds(void)
+{
+	/************ TODO: overhaul seed dispersal. This block prevents seed dispersal from running. **************/
+	printf("\nSeed dispersal during the simulation is not yet functional.\n"
+		    "Check out GitHub for updates on this feature.\n");
+	UseSeedDispersal = FALSE;
+	return;
```

Comment:
Remove redundant return.

Reply:
@fdpierson I'm going to leave this return in until we decide how to overhaul seed dispersal. I want to make it clear that this function does not work and should return immediately.